### PR TITLE
[Omega] Fix loading ISO files on Android arm-v7a

### LINKED ISSUE #25105: Loading *.iso forces close Kodi 21
### Describe the bug
Here is a clear and concise description of what the problem is:

If I try to load an *.iso file from a directly attached USB3 device Kodi forces close immediality.
Kodi 20.* was working fine - also master/nightly 22alpha is working fine. #24964 doesn't fix that. It's a different issue.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Simply play the *.iso as usual.

## Actual Behavior

Force close of Kodi.

### To Reproduce

1.  Open any *.iso file
2.
3.

### Debuglog

The debuglog can be found here:

https://paste.kodi.tv/edifeconal.kodi

The log is done with adb debugging over ssh. Also with | grep -i kodi.
Kodi 21 opens the *. iso with OnPlayMedia <file> which forces close
05-01 13:48:53.280  9015  9031 D Kodi    : **ERROR: fseeko (): Invalid argument

Kodi 22 master/nighly opens the *. iso with VideoPlayer::OpenFile: <file> which is working well

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

It's on Minix NEO U22-XJ Max. Kodi 20 was working fine. Kodi 21 can't play the *.iso even fix #24964 was applied.
Kodi 22alpha of master/nightly is working fine.

### Your Environment
Used Operating system:

 - [x ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 21

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by joseluismarti**
## Description
There is an issue on 32-bit Android devices when trying to open an ISO file from the internal storage, kodi force closes and displays the following debug message:

```
D Kodi : **ERROR: fseeko (): Invalid argument
```

The `libcdio` library is calling the `fseeko` function with a 64-bit `off_t` argument and this function is not available in Android API 21, was added in 24. 

The fix would be to target 24 as done in v22.

I propose a workaround to continue at level 21 in v21: disable `iso9660pp` so that `libudfread` is used to open ISO files, similar to how network ISO files are opened.

## How has this been tested?
Tested to open and play a DVD ISO file on a 32-bit Android device.

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by joseluismarti** [comment]
The builder has ignored the argument: 

```
-- ISO9660PP enabled: Yes
```

Locally it has worked, any advice?

**Comment by thexai** [comment]
No idea, ping @fuzzard

**Comment by thexai** [comment]
Maybe `False` instead `OFF` ?

**Comment by joseluismarti** [comment]
> Maybe `False` instead `OFF` ?

Format is correct (also seen [here](https://github.com/xbmc/xbmc/blob/master/cmake/README.md#build-options)) as it works locally. Maybe when configuring it's redefined, better wait for ping

File: cmake/scripts/android/ArchSetup.cmake [hunk 1]
```diff
@@ -17,6 +17,7 @@ else()
   if(CPU STREQUAL armeabi-v7a)
     set(ARCH arm)
     set(NEON True)
+    set(ENABLE_ISO9660PP OFF)
```
Comment (howie-f) [inline cmake/scripts/android/ArchSetup.cmake:20]: maybe needs this syntax?
https://github.com/xbmc/xbmc/blob/b8fe3f314de594c09dc7c94cd5bed7ffc0fe59c2/cmake/scripts/android/ArchSetup.cmake#L38

**Comment by joseluismarti** [comment]
Works! thank you both

File: cmake/scripts/android/ArchSetup.cmake (hunk 1 above)
Comment (joseluismarti) [reply cmake/scripts/android/ArchSetup.cmake]: Perfect, thanks

**Comment by fuzzard** [comment]
My concern is what is disabling the iso9660p library actually going to do?

I dont understand how this is a new problem in Omega over Nexus. Nothing has changed regarding libcdio in Omega as far as i could see. is there something specifically changed in Omega that somehow caused this? Does this even work in Nexus?

**Comment by joseluismarti** [comment]
I've checked that the last release that opens an iso file is 18.9, at that time libudfread was used.

From 19.0 onwards crashes with the invalid argument message in `fseeko`.

libcdio was added in 19.0 (#16262)

**Review by thexai — APPROVED** [review]
Tested on Sony TV (arm-v7a): tested DVD ISO from USB storage and SMB. 

Also tested Blu-Ray BDMV folder and BD ISO from SMB for possible regressions. All is working fine.

**Comment by howie-f** [comment]
> Tested on Sony TV (arm-v7a): tested DVD ISO from USB storage and SMB.

apart from fuzzard's concerns.. wasn't the error only happening when trying to open an ISO from internal storage? not SMB/USB

**Comment by thexai** [comment]
USB also was crashing for me in this Sony TV with Play Store version

**Comment by lrusak** [comment]
What about making the UDF implementation higher priority? 

https://github.com/xbmc/xbmc/blob/161ccd09dca7980500bccc06e5b6012c11ea9a8b/xbmc/filesystem/FileDirectoryFactory.cpp#L158

**Comment by joseluismarti** [comment]
> What about making the UDF implementation higher priority?

Could also be fixed in this way, but it would affect all platforms. To reduce the risks better the current solution that only affects Android arm-v7a.

**Comment by lrusak** [comment]
> > What about making the UDF implementation higher priority?
> 
> 
> 
> Could also be fixed in this way, but it would affect all platforms. To reduce the risks better the current solution that only affects Android arm-v7a.

I would check if that solves the problem.

Also has this change been made to master? I see it's PR'd to the Omega branch.

**Comment by joseluismarti** [comment]
> Also has this change been made to master? I see it's PR'd to the Omega branch.

Master targets Android API 24, at this level `fseeko` is "fully" available and therefore `libcdio` works properly, no change is necessary.

**Comment by lrusak** [comment]
> > Also has this change been made to master? I see it's PR'd to the Omega branch.
> 
> 
> 
> Master targets Android API 24, at this level `fseeko` is "fully" available and therefore `libcdio` works properly, no change is necessary.
> 
> 

Could libcdio be patched to fix the issue?

**Comment by joseluismarti** [comment]
> Could libcdio be patched to fix the issue?

A 64-bit fseeko function or something similar should be implemented, but at the moment since it doesn't affect Master and there is a workaround that works for v21 I don't think it's worth.

**Comment by fritsch** [comment]
@lrusak  reasoning @joseluismarti  fine with you? Given that it is about v21 stable?

**Comment by fritsch** [comment]
Yeah - also not my usecase. So whatever was the winner of the discussion should make it.

**Comment by howie-f** [comment]
> So whatever was the winner of the discussion should make it.

there was no real winner. i just raised the concern that disabling iso9660pp could probably break playback of physical dvd‘s… but android is also not my use case nor can i test dvd playback anymore.

not an easy decision. maybe merge as is and revert if shit happens 🤷

**Comment by joseluismarti** [comment]
I also don't use this feature just to search for the fix, in the tests I've done I can see the content of the iso file and the playback also works. 

The reporter and someone else confirm that it works, apparently well.

**Comment by fritsch** [comment]
Good - if we all don't know. Let's do what @howie-f suggested.

**Comment by fritsch** [comment]
Thanks everyone involved.

**Pull request merged**